### ISSUE TITLE
Percolator not supporting nested aggregations

### ISSUE THREAD

**Issue by tarass** [elastic/elasticsearch#16711]
I wanted to aggregate my percolation results with a nested aggregation. However, it appears that code path isn't implemented. Martijn (@martijnvg) you appeared to know the issue.

PercolateContext.java
```
@Override
public MapperService.SmartNameObjectMapper smartNameObjectMapper(String name) {
throw new UnsupportedOperationException();
}
```
Is there support for this coming soon? It would be nice utility that would help avoid duplicating percolators.

Here is the script to reproduce the problem

```
PUT /classify
{
  "mappings":{
    ".percolator":{
      "properties": {
        "query" : {
          "type" : "object",
          "enabled" : false
        },
        "isa":{
          "type": "nested",
          "properties":{
            "feature":{
              "type": "string"
            },
            "id": {
              "type": "string"
            }
          }
        }
      }
    },
    "thing":{
      "properties":{
        "name":{
          "type":"string"
        }
      }
    }
  }
}

POST classify/.percolator/1234
{
  "query": {
    "match": {
      "name": "arts"
    }
  },
  "type": "thing",
  "isa":[{
    "feature": "Category",
    "id": "theatre"
  },
  {
    "feature": "Ambience",
    "id": "classy"
  }
  ]
}

GET /classify/thing/_percolate
{
  "doc": {
    "name": "Academy of Arts"
  },
  "aggs":{
    "isa":{
      "nested": {
        "path": "isa"
      },
      "aggs": {
        "features": {
          "terms": {
            "field": "isa.feature"
          },
          "aggs": {
            "kinds": {
              "terms": {
                "field": "isa.id"
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by martijnvg** [comment]
@tarass thanks for reporting. This caused by a bug.

**Comment by martijnvg** [comment]
@tarass This issue will be fixed in the upcoming 5.0 release. The bug has been fixed by the refactoring done in #16349.
